src/Sentry.Profiling/SampleProfilerSession.cs

```diff
@@ -52,15 +53,43 @@ private SampleProfilerSession(SentryStopwatch stopwatch, EventPipeSession sessio
 
     public static SampleProfilerSession StartNew(IDiagnosticLogger? logger = null)
     {
-        var client = new DiagnosticsClient(Process.GetCurrentProcess().Id);
-        var session = client.StartEventPipeSession(Providers, requestRundown: false, CircularBufferMB);
-        var stopWatch = SentryStopwatch.StartNew();
-        var eventSource = TraceLog.CreateFromEventPipeSession(session, TraceLog.EventPipeRundownConfiguration.Enable(client));
+        try
+        {
+            var client = new DiagnosticsClient(Process.GetCurrentProcess().Id);
+
+            // Note: StartEventPipeSession() can time out after 30 seconds on resource constrained systems.
+            // See https://github.com/dotnet/diagnostics/blob/991c78895323a953008e15fe34b736c03706afda/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcClient.cs#L40C52-L40C52
+            var session = client.StartEventPipeSession(Providers, requestRundown: false, CircularBufferMB);
 
-        // Process() blocks until the session is stopped so we need to run it on a separate thread.
-        Task.Factory.StartNew(eventSource.Process, TaskCreationOptions.LongRunning);
+            var stopWatch = SentryStopwatch.StartNew();
+            var eventSource = TraceLog.CreateFromEventPipeSession(session, TraceLog.EventPipeRundownConfiguration.Enable(client));
 
-        return new SampleProfilerSession(stopWatch, session, eventSource, logger);
+            // Process() blocks until the session is stopped so we need to run it on a separate thread.
+            Task.Factory.StartNew(eventSource.Process, TaskCreationOptions.LongRunning);
```

Comment:
Do we need to observe any exception of this and log? I think right this it'll end up on UnobservedExceptionHandler

Reply:
it's logged below on line 74

I've added a test case `Profiler_ThrowingOnSessionStartup_DoesntBreakSentryInit` to make sure an exception thrown inside profiler session init doesn't break Sentry init

Reply:
Sorry I don't get it (line 74 is above this). But I mean if the callback running concurrently throws (eventSource.Process`, if that throws? We can `ContinuesWith` at the end of `StartNew` for example, or have a callback with a trycatch before we call eentSource.Process?

Reply:
Oh, I get what you mean now. I was confused at first because this bit of code hasn't actually changed so I thought you're talking about sth else.

In case `Process()` throws: yes, I think it'd get handled by UnobservedExceptionHandler.

I've added ContinueWith()